metagraph/src/cli/align.cpp

```diff
@@ -416,26 +373,26 @@ int align_to_graph(Config *config) {
         for (const auto &file : files) {
             logger->trace("Map sequences from file {}", file);
 
-            map_sequences_in_file(file, *graph, dbg, *config, timer,
-                                  &thread_pool, &print_mutex);
+            map_sequences_in_file(file, *graph, *config, timer, &thread_pool, &print_mutex);
         }
 
         thread_pool.join();
 
         return 0;
     }
 
-    DBGAlignerConfig aligner_config = initialize_aligner_config(graph->get_k(), *config);
+    DBGAlignerConfig aligner_config = initialize_aligner_config(*config);
 
     for (const auto &file : files) {
         logger->trace("Align sequences from file {}", file);
         seq_io::FastaParser fasta_parser(file, config->forward_and_reverse);
+        bool is_reverse_complement = false;
 
         Timer data_reading_timer;
 
-        std::ostream *out = config->outfbase.size()
-            ? new std::ofstream(config->outfbase)
-            : &std::cout;
+        std::shared_ptr<std::ostream> out{ std::shared_ptr<std::ostream>{}, &std::cout };
+        if (config->outfbase.size())
+            out = std::make_shared<std::ofstream>(config->outfbase);
```

Comment:
hmm....I never explicitly make copies of this, but I guess it would be safer to use unique_ptr instead since I never modify this.

or, I could just make it only output to stdout and forget about writing to a file

Reply:
I just replaced it with a unique_ptr

Reply:
This will work, unique_ptr doesn't allow copying, where potential race conditions could happen.